File: Python/Algorithm/baekjoon/10866_덱.py
# baekjoon source = "https://www.acmicpc.net/problem/10866"
import sys
from collections import deque

d = deque()

print(dir(d))
N = int(sys.stdin.readline())
q = []

for _ in range(N):
    cmd = sys.stdin.readline().split()

    if len(cmd) == 2:
        if cmd[0] == 'push_back':
            q.append(cmd[1])
        else:
            q.insert(0, cmd[1])
    else:
        if cmd[0] == 'pop_front':
            if len(q) == 0:
                print('-1')
            else:
                print(q.pop(0))
        elif cmd[0] == 'pop_back':
            if len(q) == 0:
                print('-1')
            else:
                print(q.pop(len(q)-1))
        elif cmd[0] == 'size':
            print(len(q))
        elif cmd[0] == 'empty':
            if q:
                print('0')
            else:
                print('1')
        elif cmd[0] == 'front':
            if q:
                print(q[0])
            else:
                print('-1')
        else:
            if q:
                print(q[len(q)-1])
            else:
                print('-1')

File: Java/baekjoon/화목한스터디/src/큐/_10866_덱.java
//baekjoon source = "https://www.acmicpc.net/problem/10866"
package Å¥;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Deque;
import java.util.LinkedList;
import java.util.StringTokenizer;

public class _10866_µ¦ {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		Deque<Integer> deq = new LinkedList<Integer>();
		int n = Integer.parseInt(br.readLine());
		StringBuilder sb = new StringBuilder();
		
		for (int i = 0; i < n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			switch (st.nextToken()) {
			case "push_back":
				deq.add(Integer.parseInt(st.nextToken()));
				break;
			case "push_front":
				deq.addFirst(Integer.parseInt(st.nextToken()));
				break;
			case "front":
				if (deq.isEmpty()) {
					sb.append(-1+"\n");
				}else {
					sb.append(deq.peek()+"\n");
				}
				break;
			case "back":
				if (deq.isEmpty()) {
					sb.append(-1+"\n");
				}else {
					sb.append(deq.peekLast()+"\n");
				}
				break;
			case "size":
				sb.append(deq.size()+"\n");
				break;
			case "empty":
				if (deq.isEmpty()) {
					sb.append(1+"\n");
				}else {
					sb.append(0+"\n");
				}
				break;
			case "pop_front":
				if (deq.size() == 0) {
					sb.append(-1+"\n");
				}else {
					sb.append(deq.pollFirst()+"\n");
				}
				break;
			case "pop_back":
				if (deq.size() == 0) {
					sb.append(-1+"\n");
				}else {
					sb.append(deq.pollLast()+"\n");
				}
				break;
			}
		}
		
		bw.write(sb+"\n");
		bw.flush();
	}
}
